tests/test_display.py

```diff
@@ -81,6 +96,23 @@ def test_display_default(self, mock_put_text):
                                          position=(224, 294),
                                          thickness=self.thickness)
 
+    @patch('sense.display.DisplayClassnameOverlay._display_class_name')
+    def test_display_default_within_duration(self, mock_display_class_name):
+        test_display = base_display.DisplayClassnameOverlay(thresholds={'Dabbing': 0.5}, duration=.1)
+        start_time = time.perf_counter()
+        test_display.display(self.img, {'sorted_predictions': [['Dabbing', 0.6]]})
+
+        call_count = 0
+        while time.perf_counter() - start_time <= test_display.duration:
```

Comment:
Here you're actually waiting for the time to pass, which slows your tests down by 1 second. A way better approach is to mock `time.perf_counter` and artificially increase the time.
However, mocking `time` is complicated, which is why you probably need to extract the `now = time.perf_counter()` line from `display()` into its own method so you can mock that.